File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int ans = 0;
        int count = 0;
        for(int i=0; i<s.size(); i++) {
            if(s[i]==')' && count == 0)
                ans++;
            else if(s[i]=='(')
                count++;
            else 
                count--;
        }
        return ans+count;
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int ans = 0;
        int count = 0;
        for(char c : s.toCharArray()) {
            if(count == 0 && c == ')')
                ans++;
            else if(c == ')')
                count--;
            else 
                count++;
        }
        ans = ans+count;
        return ans;
    }
}